lte/gateway/c/core/oai/common/common_defs.h

```diff
@@ -51,11 +51,21 @@
 #define COUNT_OF(x) \
   ((sizeof(x) / sizeof(0 [x])) / ((size_t)(!(sizeof(x) % sizeof(0 [x])))))
 
+#ifndef __cplusplus
 #define PARENT_STRUCT(cOnTaiNeD, TyPe, MeMBeR)                    \
   ({                                                              \
     const typeof(((TyPe*)0)->MeMBeR)* __MemBeR_ptr = (cOnTaiNeD); \
     (TyPe*)((char*)__MemBeR_ptr - OFFSET_OF(TyPe, MeMBeR));       \
   })
+#endif
+
+#ifdef __cplusplus
+#define PARENT_STRUCT(cOnTaiNeD, TyPe, MeMBeR)                    \
```

Comment:
Is e.g. `cOnTaiNeD` some kind of naming convention for C++ macros?

Reply:
The Magma Code Conventions (for C) can be found [in the Wiki](https://github.com/magma/magma/wiki/Contributing-Code-Conventions#c) and are referring to the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html).

I was not able to find anything advising this type of naming on first glance. So imho, this should be either cleaned up (_preferred_) or a direct link to the section demanding that formatting should be given.

Reply:
My intent was to make it easy to see what i had and had not changed in this hairy macro.  So changing names was something I decided not to do in this PR (there were some 0.o over the naming conventions here).  I do not know WHY this form of naming conventions were adopted, nor what the rules are - but OAI appears to be "consistent" across their macros [examples can be found here](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/magma/magma%24+%23define.*%28&patternType=regexp).  Happy to rename the fields / fixup naming in this PR if you prefer, however.

Reply:
Ok. I understand that no one knows, why it was introduced in first place, but it is widely used in OAI.
So renaming might just be out-of-scope in this PR to keep it consistent with the rest of the component.

Reply:
Once this PR is merged I'll float [this renaming PR](https://github.com/electronjoe/magma/commit/415bb8e81c0e90f60dbbba7b37dde53f7a7d738a)

Reply:
This was done like this, in a so silly way to avoid collisions with whatever,  whatever would be the (consustent) naming convention.